firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -124,6 +134,51 @@ public Thread newThread(Runnable r) {
     this.utils = utils;
     this.iidStore = iidStore;
     this.fidGenerator = fidGenerator;
+    preventParallelUsageOfIncompatibleVersionOfInstanceIdSDk();
+  }
+
+  private void preventParallelUsageOfIncompatibleVersionOfInstanceIdSDk() {
+    if (!isIIDVersionCompatible()) {
+      throw new IllegalStateException(
+          "FirebaseInstallations will not work correctly with current version of Firebase "
+              + "Instance ID SDK. Please update your Firebase Instance ID version.");
+    }
+  }
+
+  /**
+   * Returns TRUE iff IID SDK is not used along with FIS SDK or IID SDK version is &gt than 20.1.1,
+   * otherwise returns FALSE.
+   */
+  private boolean isIIDVersionCompatible() {
+    try {
+      final Class<FirebaseInstanceId> firebaseInstanceIdClass = FirebaseInstanceId.class;
+    } catch (NoClassDefFoundError exception) {
+      // It is OK if there is no IID SDK at all.
+      return true;
+    }
+    String iidVersion = getIIDVersion();
+    // IID SDK version is incompatible if it is older than the IID SDK version with FIS
+    // dependency(i.e 20.1.1).
+    if (TextUtils.isEmpty(iidVersion)
```

Comment:
afaict this check is not future-proof as iid won't have a properties file once released from github(something @VinayGuthal is working on)

fyi @ashwinraghav

Reply:
I wasn't aware of this. Clients moving from IID to directly depend on FIS have been requesting such a check to avoid running into multiple firebase identifiers. Without such a check in place we cant be sure that there is only one identifier.

Reply:
Question:
Why are we making changes to FirebaseInstanceId, when the plan is to deprecate this SDK as soon as possible?
Can't we just leave FirebaseInstanceId the way it is right now and focus on moving all Firebase SDKs to directly depend on FIS?